models/selective_triggers.go

```diff
@@ -0,0 +1,381 @@
+	Name    any   `json:"name,omitempty" yaml:"name,omitempty"`
+}
+
+func (pushItem PushGitEventTriggerItem) toString() string {
+	return fmt.Sprintf("PushGitEventTriggerItem{Branch: %v, CommitMessage: %v, ChangedFiles: %v}", pushItem.Branch, pushItem.CommitMessage, pushItem.ChangedFiles)
+}
+
+func (pullRequestItem PullRequestGitEventTriggerItem) toString() string {
+	draftEnabled := defaultDraftPullRequestEnabled
+	if pullRequestItem.DraftEnabled != nil {
+		draftEnabled = *pullRequestItem.DraftEnabled
+	}
+	return fmt.Sprintf("PullRequestGitEventTriggerItem{DraftEnabled: %v, SourceBranch: %v, TargetBranch: %v, Label: %v, Comment: %v, CommitMessage: %v, ChangedFiles: %v}", draftEnabled, pullRequestItem.SourceBranch, pullRequestItem.TargetBranch, pullRequestItem.Label, pullRequestItem.Comment, pullRequestItem.CommitMessage, pullRequestItem.ChangedFiles)
+}
+
+func (tagItem TagGitEventTriggerItem) toString() string {
+	return fmt.Sprintf("TagGitEventTriggerItem{Name: %v}", tagItem.Name)
+}
+
+// UnmarshalYAML implements the yaml.Unmarshaler interface for Triggers, allowing
+// additional validation of the triggers YAML configuration.
+func (triggers *Triggers) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	var triggersConfig map[string]any
+	if err := unmarshal(&triggersConfig); err != nil {
+		return fmt.Errorf("'triggers': should be a map with 'push', 'pull_request' and 'tag' keys")
+	}
+
+	if err := ensureKeys(triggersConfig, "push", "pull_request", "tag"); err != nil {
+		return fmt.Errorf("'triggers': %w", err)
+	}
+
+	if pushTriggersRaw, ok := triggersConfig["push"]; ok {
+		pushTriggers, err := parsePushTriggers(pushTriggersRaw)
+		if err != nil {
+			return err
+		}
+
+		triggers.PushTriggers = pushTriggers
+	}
+
+	if pullRequestTriggersRaw, ok := triggersConfig["pull_request"]; ok {
+		pullRequestTriggers, err := parsePullRequestTriggers(pullRequestTriggersRaw)
+		if err != nil {
+			return err
+		}
+
+		triggers.PullRequestTriggers = pullRequestTriggers
+	}
+
+	if tagTriggersRaw, ok := triggersConfig["tag"]; ok {
+		tagTriggers, err := parseTagTriggers(tagTriggersRaw)
+		if err != nil {
+			return err
+		}
+
+		triggers.TagTriggers = tagTriggers
+	}
+
+	return nil
+}
+
+func parsePushTriggers(pushTriggersRaw any) ([]PushGitEventTriggerItem, error) {
+	pushTriggersList, ok := pushTriggersRaw.([]any)
+	if !ok {
+		return nil, fmt.Errorf("'triggers.push': should be a list of push trigger items")
+	}
+
+	seenPushItems := map[string]int{}
+	var pushTriggers []PushGitEventTriggerItem
+	for idx, pushTriggerRaw := range pushTriggersList {
+		pushTriggerItem, err := parsePushTriggerItem(pushTriggerRaw)
+		if err != nil {
+			return nil, fmt.Errorf("'triggers.push[%d]': %w", idx, err)
+		}
+
+		pushTriggerItemStr := pushTriggerItem.toString()
+		seenIdx, ok := seenPushItems[pushTriggerItemStr]
+		if ok {
+			return nil, fmt.Errorf("'triggers.push[%d]': duplicates the %d. push trigger item", idx, seenIdx)
```

Comment:
```suggestion
			return nil, fmt.Errorf("'triggers.push[%d]': duplicates push trigger item #%d", idx, seenIdx)
```
I wanted to write `%dth`, but then `1st` and `2nd` wouldn't work and I don't think it's worth pulling in a dependency